appinventor/appengine/src/com/google/appinventor/shared/simple/ComponentDatabaseInterface.java

```diff
@@ -16,6 +20,131 @@
  * @author lizlooney@google.com (lizlooney)
  */
 public interface ComponentDatabaseInterface {
+
+  /**
+   * Simple component information: component name, its properties
+   */
+  public static class ComponentDefinition {
```

Comment:
Here is the Component class that was present in ComponentDatabase. 
Moved it here. Made public. 
Was necessary to access the old ComponentDefinition for upgrading the MockComponent. (See MockComponent.upgrade() )
Renamed it to ComponentDefinition, so it is similar to the PropertyDefinition class .